tests/basic/customer_issues_test.go

```diff
@@ -972,3 +976,174 @@ var _ = Describe("{ContainerCreateDeviceRemoval}", func() {
+		return err
+	}
+
+	return nil
+}
+
+func getPVCAccessMode(pvcName string, pvcNameSpace string) ([]corev1.PersistentVolumeAccessMode, error) {
+	accessModes := []corev1.PersistentVolumeAccessMode{}
+	pvc, err := k8sCore.GetPersistentVolumeClaim(pvcName, pvcNameSpace)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, mode := range pvc.Spec.AccessModes {
+		log.Infof("Access Mode [%v]", mode)
+		accessModes = append(accessModes, mode)
+	}
+
+	//return nil, fmt.Errorf("Failed to get details of PVC Access mode ")
+	return accessModes, nil
+}
+
+// Flush all IPTable Rules from all the Nodes Created
+func flushAllIPtableRulesOnAllNodes() {
+	for _, eachNode := range node.GetNodes() {
+		command := "iptables -F"
+		if !node.IsMasterNode(eachNode) {
+			err := runCommand(command, eachNode)
+			if err != nil {
+				log.FailOnError(err, "Failed to flush iptable rules")
+			}
+		}
+	}
+}
+
+var _ = Describe("{FADAPodRecoveryAfterBounce}", func() {
+
+	/*
+			PTX : https://purestorage.atlassian.net/browse/PWX-31647
+		Test to check if the Pod bounces and Places in the new node when iscsi port fails to Send / Receive Traffic
+		Test is Specific to FADA Volume ,
+
+	*/
+	JustBeforeEach(func() {
+		StartTorpedoTest("FADAPodRecoveryAfterBounce",
+			"Verify Pod Recovers from RO mode after Bounce",
+			nil, 0)
+	})
+	var contexts []*scheduler.Context
+	stepLog := "App Stuck in ContainerCreation State with Device Exists in the backend"
+	It(stepLog, func() {
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("fapodrecovery-%d", i))...)
+		}
+		ValidateApplications(contexts)
+		defer DestroyApps(contexts, nil)
+		defer flushAllIPtableRulesOnAllNodes()
+
+		// Wait for 2 min for pods to get settle down before proceeding further
+		time.Sleep(2 * time.Minute)
+
+		// Pick all the Volumes with RWO Status, We check if the Volume is with Access Mode RWO and PureBlock Volume
+		vols := make([]*volume.Volume, 0)
+		for _, ctx := range contexts {
+			appVols, err := Inst().S.GetVolumes(ctx)
+			log.FailOnError(err, fmt.Sprintf("error getting volumes for app [%s]", ctx.App.Key))
+
+			for _, eachVol := range appVols {
+				accessModes, err := getPVCAccessMode(eachVol.Name, eachVol.Namespace)
+				log.FailOnError(err, "Failed to get AccessModes for the volume [%v]", eachVol.Name)
+
+				for _, eachAMode := range accessModes {
+					// Validate if the Volume is Pure Volume
+					boolVol, err := Inst().V.IsPureVolume(eachVol)
+					log.FailOnError(err, "Failed to get details on the volume [%v]", eachVol.Name)
+
+					// Get Details of the volume , check if the volume is PureBlock Volume
+					pureVol, err := IsVolumeTypePureBlock(ctx, eachVol.ID)
+					log.FailOnError(err, "Failed to get details on the volume [%v]", eachVol.Name)
+
+					if eachAMode == "ReadWriteOnce" && boolVol && pureVol {
+						vols = append(vols, eachVol)
+					}
+				}
+			}
+		}
```

Comment:
How about?

```suggestion
		for _, ctx := range contexts {
		    pureVols, err := Inst().S.GetPureVolumes(ctx, "pure_block")
		    log.FailOnError(err, fmt.Sprintf("error getting pure_block volumes for app [%s]", ctx.App.Key))

		    for _, eachVol := range pureVols {
		        accessModes, err := getPVCAccessMode(eachVol.Name, eachVol.Namespace)
		        log.FailOnError(err, "Failed to get AccessModes for the volume [%v]", eachVol.Name)

		        for _, eachAMode := range accessModes {
		            if eachAMode == "ReadWriteOnce" {
		                vols = append(vols, eachVol)
		            }
		        }
		    }
		}
```

This may be beyond the scope of this PR, but we could consider modifying Inst().S.GetVolumes or Inst().S.GetPureVolumes to return a list of *volume.Volume that includes access modes.

Reply:
let me see if this can be updated in portworx.go .

Reply:
using pureVols, err := Inst().S.GetPureVolumes(ctx, "pure_block") now